superset-frontend/src/SqlLab/components/ResultSet/index.tsx

```diff
@@ -220,6 +224,15 @@ export default class ResultSet extends React.PureComponent<
       const { showSaveDatasetModal } = this.state;
       const { query } = this.props;
 
+      const dataset: ISaveableDataset = {
+        columns: query.columns as ISimpleColumn[],
+        name: query?.tab || 'Untitled',
+        dbId: 1,
+        sql: query.sql,
+        templateParams: query.templateParams,
+        schema: query.schema,
```

Comment:
if this is a query and not a dataset, can we call it something more generic like a datasource? Dataset is a different model altogether, so it's a little confusing.